storage/table/fsm/fsm.go

```diff
@@ -156,7 +156,12 @@ func (p *FSM) Open(_ <-chan struct{}) (uint64, error) {
 		p.log.Errorf("unable to register metrics for FSM: %s", err)
 	}
 
-	return readLocalIndex(db, sysLocalIndex)
+	idx, err := readLocalIndex(db, sysLocalIndex)
+	if err != nil {
+		return 0, err
+	}
+	p.metrics.applied.Store(idx)
+	return idx, err
```

Comment:
minor: since the err is already handled, explicitly returning `nil might be more readable
```suggestion
	return idx, nil
```